syntax = "proto3";
import "google/protobuf/struct.proto";
import "BlueprintCommon.proto";
option java_multiple_files = true;
package org.onap.ccsdk.cds.controllerblueprints.processing.api;


message ExecutionServiceInput {
  org.onap.ccsdk.cds.controllerblueprints.common.api.CommonHeader commonHeader = 1;
  org.onap.ccsdk.cds.controllerblueprints.common.api.ActionIdentifiers actionIdentifiers = 2;
  google.protobuf.Struct payload = 3;
}

message ExecutionServiceOutput {
  org.onap.ccsdk.cds.controllerblueprints.common.api.CommonHeader commonHeader = 1;
  org.onap.ccsdk.cds.controllerblueprints.common.api.ActionIdentifiers actionIdentifiers = 2;
  org.onap.ccsdk.cds.controllerblueprints.common.api.Status status = 3;
  google.protobuf.Struct payload = 4;
}


service BlueprintProcessingService {
  rpc process (stream ExecutionServiceInput) returns (stream ExecutionServiceOutput);
}
